pkg/controller/storagecluster/uninstall_reconciler_test.go

```diff
@@ -131,3 +137,78 @@ func TestDeleteStorageClasses(t *testing.T) {
 		}
 	}
 }
+
+func TestDeleteNodeAffinityKeyFromNodes(t *testing.T) {
+
+	for _, eachPlatform := range allPlatforms {
+		cp := &CloudPlatform{platform: eachPlatform}
+		_, reconciler, sc, _ := initStorageClusterResourceCreateUpdateTestWithPlatform(t, cp, nil)
+
+		// case1: verify deleteNodeAffinityKeyFromNodes deletes default NodeAffinityKey
+		// get eligible nodes
+		nodes := assertGetStorageClusterEligibleNodes(t, reconciler, sc)
+
+		// verify default NodeAffinityKey present on nodes
+		for _, node := range nodes.Items {
+			_, ok := node.ObjectMeta.Labels[defaults.NodeAffinityKey]
+			assert.Equal(t, ok, true)
+		}
+
+		// delete NodeAffinityKey
+		err := reconciler.deleteNodeAffinityKeyFromNodes(sc, reconciler.reqLogger)
+		assert.NoError(t, err)
+
+		// get and verify there are no eligible nodes after deleting NodeAffinityKey
+		nodes, err = reconciler.getStorageClusterEligibleNodes(sc, reconciler.reqLogger)
+		assert.NoError(t, err)
+		assert.Equal(t, 0, len(nodes.Items))
+
+		// case2: verify deleteNodeAffinityKeyFromNodes does not deletes user defined key
+		// create user defined key and val and apply it on sc and nodes
+		fakeKey, fakeVal := "fakeKey", "fakeVal"
+		sc.Spec.LabelSelector = metav1.AddLabelToSelector(&metav1.LabelSelector{}, fakeKey, fakeVal)
+
+		// get all nodes
+		nodes = &corev1.NodeList{}
+		err = reconciler.client.List(context.TODO(), nodes)
+		assert.NoError(t, err)
+		assert.NotEqual(t, 0, len(nodes.Items))
+
+		// Add labels on nodes
+		for _, node := range nodes.Items {
+			new := node.DeepCopy()
+			new.ObjectMeta.Labels[fakeKey] = fakeVal
+
+			oldJSON, err := json.Marshal(node)
+			assert.NoError(t, err)
+
+			newJSON, err := json.Marshal(new)
+			assert.NoError(t, err)
+
+			patch, err := strategicpatch.CreateTwoWayMergePatch(oldJSON, newJSON, node)
+			assert.NoError(t, err)
+
+			err = reconciler.client.Patch(context.TODO(), &node, client.RawPatch(types.StrategicMergePatchType, patch))
+			assert.NoError(t, err)
+		}
+
+		// verify there are eligible nodes after adding user defined key
+		assertGetStorageClusterEligibleNodes(t, reconciler, sc)
+
+		// delete NodeAffinityKey
+		err = reconciler.deleteNodeAffinityKeyFromNodes(sc, reconciler.reqLogger)
+		assert.NoError(t, err)
+
+		// verify deleteNodeAffinityKeyFromNodes does not delete user defined key
+		assertGetStorageClusterEligibleNodes(t, reconciler, sc)
+	}
+}
+
+func assertGetStorageClusterEligibleNodes(
```

Comment:
```suggestion
func getStorageClusterEligibleNodes(
```

Reply:
If you need to assert values of  "node", do it after function call.

Reply:
Deleted the func as I have mainly created the func for the assert and we already have a func for getting nodes